mw-webapp/src/component/editableText/EditableText.tsx

```diff
@@ -50,15 +57,24 @@ export const EditableText = <T extends string | number>(props: EditableTextProps
     }
   };
 
+  /**
+   * Check type of coming value and convert it to Number if need to use input with type "number"
+   */
+  const setValue = (value: string) => {
+    const number = Number(value);
+    const updatedValue = typeof props.text === "string" ? value : number;
+    setText(updatedValue as T);
+  };
+
   /**
    * Render input
    */
   const renderInput = () => (
     <Input
-      type="text"
+      type={typeof props.text === "string" ? "text" : "number"}
```

Comment:
add new prop type: 'text' | 'number' instead of runtime type check. That why we are using typescript

Reply:
Changed